src/Entity/Enemy.cpp

```diff
@@ -0,0 +1,159 @@
+#include "Enemy.h"
+
+#include <raymath.h>
+
+#include "Globals.h"
+#include "Map/Map.h"
+#include "Entity/Character.h"
+#include "Rendering/Components/AnimationComponent.h"
+#include "Rendering/Components/TextureComponent.h"
+#include "Rendering/RenderUtils.h"
+
+namespace game
+{
+//QUESTION: would i declare all private static functions with forward declaratiobn?
+
+static void SpawnNewEnemy(flecs::iter& iter);
+flecs::entity gGoblinPrefab{};
+
+void Enemy::InitEnemy(flecs::world& ecs)
+{
+	const TransformComponent transformComponent{Vector2{500, 500}, Vector2Scale(Vector2{ENEMY_WIDTH, ENEMY_HEIGHT}, ENTITY_SCALE)};
+	
+	gGoblinPrefab = ecs.prefab<>("goblinPrefab")
+							.add<EnemyTag>()
+							.override<TransformComponent>()
+							.override<VelocityComponent>()
+							.set<CollisionComponent>({ENEMY_WIDTH * ENTITY_SCALE, ENEMY_HEIGHT * ENTITY_SCALE})
+							.set<VelocityComponent>({{0, 0}, ENEMY_SPEED})
+							.set<SpriteComponent>({renderUtils::LoadMyTexture("characters/goblin_spritesheet_16x16_8x2.png"),
+												   {0, 0, ENEMY_WIDTH, ENEMY_HEIGHT},
+												   {transformComponent.mScale.x, transformComponent.mScale.x},
+												   {0, 0},
+												   5})
+							.emplace<AnimationComponent>(5, 0.1f, 0.0f, 0)
+							.emplace<AnimationStateComponent>(AnimationName::IDLE);
```

Comment:
The issue, when enemies get faster and faster, is because of how the prefab is set up.
By default, when adding a component with `.emplace`/`.set`/`.add`, that component is gonna be shared between all instances of that prefab.
So in this instance, they were all sharing the same `SpriteComponent`, `AnimationStateComponent`, etc... So then in `UpdateAnimationFrame`, every enemy was updating the same source for the sprite component. So they were all sharing the same frame as well, same state, etc...

```suggestion
							.override<TransformComponent>()
							.set<CollisionComponent>({ENEMY_WIDTH * ENTITY_SCALE, ENEMY_HEIGHT * ENTITY_SCALE})
							.set_override<VelocityComponent>({{0, 0}, ENEMY_SPEED})
							.set_override<SpriteComponent>({renderUtils::LoadMyTexture("characters/goblin_spritesheet_16x16_8x2.png"),
												   {0, 0, ENEMY_WIDTH, ENEMY_HEIGHT},
												   {transformComponent.mScale.x, transformComponent.mScale.x},
												   {0, 0},
												   5})
							.emplace_override<AnimationComponent>(5, 0.1f, 0.0f, 0)
							.emplace_override<AnimationStateComponent>(AnimationName::IDLE);
```